lazylru.go

```diff
@@ -300,20 +300,28 @@ func (lru *LazyLRU[K, V]) Get(key K) (V, bool) {
 	// We only want to shuffle this item if it is far enough from the front that
 	// it is at risk of being evicted. This will save us from exclusive locking
 	// 75% of the time.
-	if lru.shouldBubble(pqi.index) {
-		if !locked {
-			lru.lock.Lock()
-			locked = true
-		}
-		// double check because someone else may have shuffled
-		if lru.shouldBubble(pqi.index) {
-			lru.items.update(pqi, atomic.AddUint64(&(lru.itemIx), 1))
-			lru.stats.Shuffles++
+	if !locked {
+		lru.lock.RLock()
+		maybeShould := lru.shouldBubble(pqi.index)
+		lru.lock.RUnlock()
+		if !maybeShould {
+			atomic.AddUint32(&lru.stats.KeysReadOK, 1)
+			return qi.value, ok
 		}
 	}
-	if locked {
-		lru.lock.Unlock()
+
+	if !locked {
+		lru.lock.Lock()
+		// locked = true  // ineffectual
```

Comment:
Should the comment be removed or uncommented?

Reply:
This is currently ineffectual because we are no longer checking `locked` below. However, if this function is ever refactored, the refactorer would want to uncomment that line because the `locked` boolean is likely to be used and I didn't want it to be forgotten. Or worse, for someone to think that this call to `Lock()` is somehow special. Breadcrumbs for our future selves.